lib/dao.js

```diff
@@ -2356,6 +2566,184 @@ DataAccessObject.prototype.unsetAttribute = function unsetAttribute(name, nullif
 };
 
 /**
+ * Replace set of attributes.
+ * Performs validation before replacing.
+ *
+ * @trigger `validation`, `save` and `update` hooks
+ * @param {Object} data Data to replace
+ * @param {Object} [options] Options for replace
+ * @param {Function} cb Callback function called with (err, instance)
+ */
+
+DataAccessObject.prototype.replace = function replace(data, options, cb) {
```

Comment:
> `DataAccessObject.replaceById = function(id, data, options, cb)`

Exactly.

> `DataAccessObject.prototype.replace = function replace(data, options, cb) {`

The signature is OK, but the name should be changed to `replaceAttributes` to match `updateAttributes` (or `patchAttributes`). The name is not perfect, but I don't have any counter-proposal that would be significantly better, so let's stick with `replaceAttributes` for consistency.